### ISSUE TITLE
Consumer can't get any messages after reset_offsets

### ISSUE THREAD

**Issue by andrey-puzyr** [Parsely/pykafka#851]
The following code is hangs (on the last consume() call). 

_Case definition: Kafka contains topic `test` with a lot of messages, I'm trying to move offset to some position using the `reset_offsets`. pykafka 2.7.0, Python 3.7.0_
```python
from pykafka import KafkaClient

pyclient = KafkaClient(hosts="localhost:9092")
topic = pyclient.topics[b'test']
c = topic.get_simple_consumer(
    consumer_group=b'mygroup', 
    auto_offset_reset=common.OffsetType.EARLIEST)
c.reset_offsets(partition_offsets=[(c.partitions[0], 10)])
c.commit_offsets()
c.consume()
```
New instances of consumer works as expected (starts from 11 offset).
It looks like a bug. After some manipulations I have found few workarounds:
```python
# ...
c.reset_offsets(partition_offsets=[(c.partitions[0], 10)])
c.commit_offsets()
c.consume(block=False) # return None
c.consume() # now works as expected
```
```python
# ...
c.reset_offsets(partition_offsets=[(c.partitions[0], 10)])
c.commit_offsets()
c.fetch()
c.consume() # now works as expected
```

**Comment by emmett9001** [comment]
Thank you for the detailed report, @andrey-puzyr. The `reset_offsets` function was recently [fixed](https://github.com/Parsely/pykafka/commit/605dea8f0f25fb9b4114241242fd94501d46ecc9) after some pretty serious bugs were discovered with it. Accordingly, I recommend that you test your use case with current master (https://github.com/Parsely/pykafka/commit/d8ed3dc7ae8877be5a6f0081e93223632cd5df7f or later) to see if it's been fixed.

Aside from that, there are several things to point out about your example code:
* The `fetch` call is not necessary, as it's called internally when the consumer is running
* The `commit_offsets` call is not necessary, since offsets are automatically committed. Additionally, the committed offset has no bearing on the offset of the next message returned from a `consume` call.

**Comment by andrey-puzyr** [comment]
Thank you for advices, @emmett9001 !
I'm happy to say that the issue was fixed in `2.8.0-dev.3`(i use https://github.com/Parsely/pykafka/commit/d8ed3dc7ae8877be5a6f0081e93223632cd5df7f). Also I have found a great performance improvements for the reset_offsets method. Great job! I hope `2.8.0` will be released soon..

PS.  Latest version generates tons of logs for my app, is it an expected behavior ?
`WARNING:root:Trying store offset: 10387`
